lib/plugins/amplitudePlugin.spec.ts

```diff
@@ -0,0 +1,134 @@
+import * as amplitude from '@amplitude/analytics-node'
+import type { BaseEvent, EnrichmentPlugin, Event } from '@amplitude/analytics-types'
+import { buildClient, sendGet } from '@lokalise/node-core'
+import type { FastifyInstance } from 'fastify'
+import fastify from 'fastify'
+
+import { amplitudePlugin, amplitudeTrack } from './amplitudePlugin'
+
+describe('amplitudePlugin', () => {
+  let app: FastifyInstance
+
+  beforeEach(() => {
+    app = fastify()
+  })
+
+  afterEach(async () => {
+    await app.close()
+  })
+
+  it('skip initialization if isEnabled is false', async () => {
+    const initSpy = jest.spyOn(amplitude, 'init')
+
+    await app.register(amplitudePlugin, {
+      isEnabled: false,
+    })
+
+    expect(initSpy).not.toHaveBeenCalled()
+  })
+
+  it('throw error if api key is not defined', async () => {
+    const initSpy = jest.spyOn(amplitude, 'init')
+
+    let error = null
+    try {
+      await app.register(amplitudePlugin, {
+        isEnabled: true,
+      })
+    } catch (e) {
+      error = e
+    }
+
+    expect(initSpy).not.toHaveBeenCalled()
+    expect(error).not.toBeNull()
+  })
+
+  it('Basic initialization', async () => {
```

Comment:
is this test useful?

Reply:
Yes, it was just to test that we are calling to the init method, but we are covering it on the other tests so it is useless, I will remove it